aspnetcore/performance/response-compression/samples/6.x/SampleApp/Program.cs

```diff
@@ -1,6 +1,191 @@
+#define OPT // FIRST SECOND NO_COMP OPT CUST
+#if NEVER
+#elif FIRST
+#region snippet
 var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddResponseCompression(options =>
+{
+    options.EnableForHttps = true;
+});
+
+var app = builder.Build();
+
+app.UseResponseCompression();
+
+app.MapGet("/", () => "Hello World!");
+
+app.Run();
+#endregion
+#elif SECOND
+#region snippet2
+using System.IO.Compression;
+using Microsoft.AspNetCore.ResponseCompression;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddResponseCompression(options =>
+{
+    options.EnableForHttps = true;
+    options.Providers.Add<BrotliCompressionProvider>();
+    options.Providers.Add<GzipCompressionProvider>();
+});
+
+var app = builder.Build();
+
+app.UseResponseCompression();
+
+builder.Services.Configure<BrotliCompressionProviderOptions>(options =>
+{
+    options.Level = CompressionLevel.Fastest;
+});
+
+builder.Services.Configure<GzipCompressionProviderOptions>(options =>
+{
+    options.Level = CompressionLevel.SmallestSize;
+});
```

Comment:
I think modifying `IServiceCollection` after the call to `builder.Build()` causes a runtime exception here. In this case, these two calls to `Configure` should be moved to after the call to `AddResponseCompression`.